homeassistant/components/konnected/__init__.py

```diff
@@ -237,6 +237,10 @@ def import_validator(config):
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up panel from a config entry."""
+    if SOURCE_IGNORE == entry.data:
```

Comment:
Huh, when can this happen? `entry.data` is always a dictionary. If anything, it would be `entry.source`, but then we will never set up an entry with source == ignore.

https://github.com/home-assistant/core/blob/b6e69cd370bdfd11c5bc459f7ef4aff867cf231e/homeassistant/config_entries.py#L176-L178

Reply:
If you take a look at the stack trace from the linked issue, the error is in the config flow which needs a guard:

```
  File "/usr/src/homeassistant/homeassistant/components/konnected/config_flow.py", line 294, in async_step_confirm
    entry.data[CONF_HOST] != self.data[CONF_HOST]
```

Reply:
You're correct - both in the entry.data typo and the setup. Something is funky with that stack trace too.

I think the actual bug I was attempting to fix is related to the change below on line 313. I'll rework this.